### PR TITLE
Syncronization testing after group deletingv3

### PR DESCRIPTION
**Pull request by pro-akim**
|Related issue|
|-------------|
|     #3953         |

## Description
Wazuh manager handles agent groups. 
If a group is deleted from a master cluster, there will be an instance where the agents require resynchronization (syncreq). 
If the group is deleted from a worker cluster, the cluster master will take care of reestablishing the group structure without the need for resynchronization.
This test suite tests the correct functioning of the mentioned use case.

### Updated

-test_group_sync delete_folder_master_agent_all
-test_group_sync delete_folder_worker1_agent_all
-test_group_sync delete_folder_worker2_agent_all
-test_group_sync delete_folder_master_agent_agent1
-test_group_sync delete_folder_worker1_agent1
-test_group_sync delete_folder_worker2_agent1
-test_group_sync delete_folder_master_agent_agent2
-test_group_sync delete_folder_worker1__agent2
-test_group_sync delete_folder_worker2_agent2

Each test has 3 scenarios where groups are created and agents are assigned to groups from wazuh-master, wazuh-worker1 and wazuh-worker2

---

## Testing performed

| Tester             | Test path | Jenkins | Local  | OS | Commit | Notes                |
|--------------------|-----------|---------|--------|-----|--------|----------------------|
| @pro-akim  (Developer)  | /wazuh-qa/tests/system/test_cluster/test_agent_groups          | ⚫⚫⚫ | [:green_circle:](https://github.com/wazuh/wazuh-qa/files/11706860/report1.zip)[:green_circle:](https://github.com/wazuh/wazuh-qa/files/11706861/report2.zip)[:green_circle: ](https://github.com/wazuh/wazuh-qa/files/11706862/report3.zip)|         |   6c2985d      | Nothing to highlight |
| @user (Reviewer)   |           | ⚫⚫⚫ |  ⚫⚫⚫  |        |         | Nothing to highlight |

### CODE REVIEW

**Comment by pro-akim** [comment]
|Related issue|
|-------------|
|     #3953         |

## Description
Wazuh manager handles agent groups. 
If a group is deleted from a master cluster, there will be an instance where the agents require resynchronization (syncreq). 
If the group is deleted from a worker cluster, the cluster master will take care of reestablishing the group structure without the need for resynchronization.
This test suite tests the correct functioning of the mentioned use case.


### Updated
Added new Tests:

-test_group_sync delete_folder_master_agent_in_group_all
-test_group_sync delete_folder_worker1_agent_in_group_all
-test_group_sync delete_folder_worker2_agent_in_group_all
-test_group_sync delete_folder_master_agent_in_group_agent1
-test_group_sync delete_folder_worker1_in_group_agent1
-test_group_sync delete_folder_worker2_in_group_agent1
-test_group_sync delete_folder_master_agent_in_group_agent2
-test_group_sync delete_folder_worker1_in_group_agent2
-test_group_sync delete_folder_worker2_in_group_agent2



---

## Testing performed




| Tester             | Test path | Jenkins | Local  | OS | Commit | Notes                |
|--------------------|-----------|---------|--------|-----|--------|----------------------|
| @pro-akim  (Developer)  |           | ⚫⚫⚫ | [:green_circle:](https://github.com/wazuh/wazuh-qa/files/11406179/report1.zip)[:green_circle:](https://github.com/wazuh/wazuh-qa/files/11406180/report2.zip)[:green_circle: ](https://github.com/wazuh/wazuh-qa/files/11406222/report3.zip)|         |         | Nothing to highlight |
| @user (Reviewer)   |           | ⚫⚫⚫ |  ⚫⚫⚫  |        |         | Nothing to highlight |

File: tests/system/test_cluster/test_agent_groups/test_group_sync_status.py [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+                               'files', 'ossec.conf')
+t1_configuration_parameters, t1_configuration_metadata, t1_case_ids = get_test_cases_data(test_cases_yaml)
+
+@pytest.fixture()
+def group_creation_and_assignation(metadata, target_node):
+
+    agent_ids = []
+    for agent in test_infra_agents:
+        agent_ip, agent_id, agent_name, manager_ip = register_agent(agent, test_infra_hosts[0], host_manager)
+        time.sleep(T_025)
```
Comment (juliamagan) [inline tests/system/test_cluster/test_agent_groups/test_group_sync_status.py:67]: Why is this sleep necessary?

File: tests/system/test_cluster/test_agent_groups/test_group_sync_status.py [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+    yield
+
+    for group in groups:
+        delete_agent_group(test_infra_hosts[0], group, host_manager, 'api')
+
+
+@pytest.fixture()
+def wait_end_initial_syncreq():
+    result = execute_wdb_query(query, test_infra_hosts[0], host_manager)
+    while 'syncreq' in result:
```
Comment (juliamagan) [inline tests/system/test_cluster/test_agent_groups/test_group_sync_status.py:98]: We need to add a timeout

**Review by juliamagan — CHANGES_REQUESTED** [review]
Review the clonflicts

File: tests/system/test_cluster/test_agent_groups/test_group_sync_status.py (hunk 2 above)
Comment (pro-akim) [reply tests/system/test_cluster/test_agent_groups/test_group_sync_status.py:101]: Timeout created in https://github.com/wazuh/wazuh-qa/pull/4143/commits/6c2985d1a2e92877c3130604cdbe41d533b20a75

**Pull request merged**